File: iban-validator/src/test/java/de/rwth/swc/banking/IbanValidationApi.java
package de.rwth.swc.banking;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.service.annotation.PostExchange;

public interface IbanValidationApi {

    @PostExchange("/v1/validate/iban")
    ResponseEntity<Boolean> validateIbanV1(@RequestBody String iban);

    @PostExchange("/v2/validate/iban")
    ResponseEntity<Boolean> validateIbanV2(@RequestBody Transfer transfer);

}

File: iban-validator/src/test/kotlin/de/rwth/swc/banking/IbanValidationApi.kt
package de.rwth.swc.banking

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.service.annotation.PostExchange


interface IbanValidationApi {
    @PostExchange("/api/v1/validate/iban")
    fun validateIbanV1(@RequestBody iban: String): ResponseEntity<IbanValidationResponse>

    @PostExchange("/api/v2/validate/iban")
    fun validateIbanV2(@RequestBody iban: String): ResponseEntity<IbanValidationResponse>
}
